File: TLS Inspector/Localization/UI Elements/GradientView.m
#import "GradientView.h"

@implementation GradientView

- (void) drawRect:(CGRect)rect {
    CAGradientLayer * gradient = [CAGradientLayer layer];
    gradient.frame = self.bounds;
    gradient.colors = @[(id)self.firstColor.CGColor, (id)self.secondColor.CGColor];
    gradient.startPoint = CGPointMake(0, 1);
    gradient.endPoint = CGPointMake(1, 0);
    [self.layer insertSublayer:gradient atIndex:0];
    [super drawRect:rect];
}

@end

File: TLSISwift/User Interface/GradientView.swift
import UIKit

@IBDesignable
class GradientView: UIView {

    @IBInspectable var startColor:   UIColor = .black { didSet { updateColors() }}
    @IBInspectable var endColor:     UIColor = .white { didSet { updateColors() }}

    override public class var layerClass: AnyClass { CAGradientLayer.self }

    // swiftlint:disable force_cast
    var gradientLayer: CAGradientLayer { layer as! CAGradientLayer }
    // swiftlint:enable force_cast

    func updatePoints() {
        gradientLayer.startPoint = .init(x: 1, y: 0)
        gradientLayer.endPoint = .init(x: 0, y: 1)
    }
    func updateLocations() {
        gradientLayer.locations = [0.0 as NSNumber, 1.0 as NSNumber]
    }
    func updateColors() {
        gradientLayer.colors = [startColor.cgColor, endColor.cgColor]
    }
    override public func layoutSubviews() {
        super.layoutSubviews()
        updatePoints()
        updateLocations()
        updateColors()
    }
}
